nano/node/bootstrap.cpp

```diff
@@ -2312,9 +2385,58 @@ class request_response_visitor : public nano::message_visitor
 		auto response (std::make_shared<nano::frontier_req_server> (connection, std::unique_ptr<nano::frontier_req> (static_cast<nano::frontier_req *> (connection->requests.front ().release ()))));
 		response->send_next ();
 	}
-	void node_id_handshake (nano::node_id_handshake const &) override
+	void node_id_handshake (nano::node_id_handshake const & message_a) override
 	{
-		assert (false);
+		if (connection->node->config.logging.network_node_id_handshake_logging ())
+		{
+			connection->node->logger.try_log (boost::str (boost::format ("Received node_id_handshake message from %1%") % connection->remote_endpoint));
+		}
+		if (message_a.query)
+		{
+			boost::optional<std::pair<nano::account, nano::signature>> response (std::make_pair (connection->node->node_id.pub, nano::sign_message (connection->node->node_id.prv, connection->node->node_id.pub, *message_a.query)));
+			assert (!nano::validate_message (response->first, *message_a.query, response->second));
+			auto cookie (connection->node->network.tcp_channels.assign_syn_cookie (connection->remote_endpoint));
+			nano::node_id_handshake response_message (cookie, response);
+			auto bytes = response_message.to_bytes ();
+			connection->socket->async_write (bytes, [connection = connection](boost::system::error_code const & ec, size_t size_a) {
```

Comment:
`bytes` needs to be captured by the lambda so that it remains in scope for asio

Reply:
Good catch

Reply:
Updated